### PR TITLE
ControlNet for DirectML

### PR DESCRIPTION
**Pull request by oyazdanb**
Adding controlnet example for DirectML.

### CODE REVIEW

**Review by lucnev — APPROVED** [review]

File: examples/directml/controlnet/controlnet-canny-test_sandraOh.png
Comment (hisham-hchowdhu) [inline examples/directml/controlnet/controlnet-canny-test_sandraOh.png:1]: its a generated file?

Comment (oyazdanb) [reply examples/directml/controlnet/controlnet-canny-test_sandraOh.png:1]: yes. it is a generated file. I thought we could keep one example.

File: examples/directml/controlnet/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# ControlNet Optimization with DirectML <!-- omit in toc -->
+
+This sample shows how to optimize [Control Net](https://huggingface.co/lllyasviel/sd-controlnet-openpose) with [Stable Diffusion v1-5](https://huggingface.co/runwayml/stable-diffusion-v1-5) to run with ONNX Runtime and DirectML.
+
```
Comment (hisham-hchowdhu) [inline examples/directml/controlnet/README.md:4]: some of the controlnet scripts that you use from different github, good to put reference from them here - providing the location and mentioning "heavily used the scripts from here modifying to adapt for Olive environment"

File: examples/directml/controlnet/controlnet-canny-test_sandraOh.png
Comment (hisham-hchowdhu) [reply examples/directml/controlnet/controlnet-canny-test_sandraOh.png:1]: sure

**Review by hisham-hchowdhu — APPROVED** [review]

**Pull request merged**